### DISCUSSION TITLE
Can't reverse proxy my Navidrome correctly

### DISCUSSION THREAD

**Discussion by CDSWambo** [navidrome/navidrome#3552] [Q&A]
Hello, for the last 2 hours I've tried to reverse proxy my navidrome installation using apache2 to access it over the internet, but I can't seem to make it work correctly. Now the reverse proxy "works" but I get a "server communcation error" on Navidrome. Both my services are up and running. I am attaching screenshots of my config files and the error
<img width="959" alt="Immagine 2024-12-17 150410" src="https://github.com/user-attachments/assets/ba8ba64c-3087-4663-b1c5-3a5fa75e5991" />
<img width="590" alt="Immagine 2024-12-17 150603" src="https://github.com/user-attachments/assets/d18b7ee5-861f-4b4b-9c65-ef39357fdedc" />
<img width="736" alt="Immagine 2024-12-17 150630" src="https://github.com/user-attachments/assets/a8ce1061-703c-4c17-bd68-60f315ca9884" />
<img width="372" alt="Immagine 2024-12-17 150704" src="https://github.com/user-attachments/assets/9c132b28-e32d-4ee2-9566-a650ccbb0205" />
<img width="404" alt="Immagine 2024-12-17 150724" src="https://github.com/user-attachments/assets/6e2317a7-813e-4513-8090-0b840ca49139" />

Don't know what else I could do. Tried also Caddy but absolutely nothing there

**Comment by tarator** [comment]
This is my Navidrome Apache reverse proxy configuration which works pretty well.
As you can see it's with Letsencrypt SSL (autoconfigured by certbot) which you might not need.
But take a look at websocket configuration. This might be your problem.


```
<IfModule mod_ssl.c>
    <VirtualHost *:443>
        ServerName navidrome.example.com
        ServerAdmin office@example.com

        RequestHeader set X-Forwarded-Proto "https"
        # allow websockets
        RewriteEngine On
        RewriteCond %{HTTP:Connection} Upgrade [NC]
        RewriteCond %{HTTP:Upgrade} =websocket [NC]
        RewriteRule /(.*)           wss://127.0.0.1:4533/$1 [P,L]

        ProxyPass / http://127.0.0.1:4533/ retry=1
        ProxyPassReverse / http://127.0.0.1:4533/
        ProxyPreserveHost On
        ProxyRequests Off


        SSLCertificateFile /etc/letsencrypt/live/navidrome.example.com/fullchain.pem
        SSLCertificateKeyFile /etc/letsencrypt/live/navidrome.example.com/privkey.pem
        Include /etc/letsencrypt/options-ssl-apache.conf
    </VirtualHost>
</IfModule>
```

**Reply by CDSWambo** [reply]
I'll give it a shot trying to adapt this with the websockets to my case and I'll let you know asap. Thank you!
